apps/examples/mediastreamer/mediastreamer_main.cpp

```diff
@@ -0,0 +1,237 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/init.h>
+#include <apps/platform/cxxinitialize.h>
+#include <tinyalsa/tinyalsa.h>
```

Comment:
We can remove some headers.
tinyalsa.h
fcntl.h
mediaplayer & observer.h
memory ?? -> checking needed

Reply:
Currently, I cannot remove the includes.
(1) fcntl.h : This has some issue. Without this header file, wifi_manager.h occurs build error about 'int8_t', .... . Actually this is wifi_manager.h issue.
(2) tinyalsa.h : Currently, we are using PCM_FORMAT_S16_LE enum value. So cannot remove now.
(3) memory : This is used for shared_ptr and unique_ptr.

MediaPlayer and Observer will be used next time. (I have to add SocketInputStream, ...)
So I prepared them now.